From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Daniel Bradley <daniel@pulumi.com>
Date: Wed, 22 May 2024 13:33:27 +0100
Subject: [PATCH] Set function app client_affinity_enabled field

However, this doesn't appear in the schema and therefore likely has no effect and we should consider removing this patch.

diff --git a/internal/services/web/function_app_slot_resource.go b/internal/services/web/function_app_slot_resource.go
index baaf28b822..b33a0a6364 100644
--- a/internal/services/web/function_app_slot_resource.go
+++ b/internal/services/web/function_app_slot_resource.go
@@ -259,6 +259,7 @@ func resourceFunctionAppSlotCreate(d *pluginsdk.ResourceData, meta interface{})
 
 	appServicePlanID := d.Get("app_service_plan_id").(string)
 	enabled := d.Get("enabled").(bool)
+	clientAffinityEnabled := d.Get("client_affinity_enabled").(bool)
 	httpsOnly := d.Get("https_only").(bool)
 	dailyMemoryTimeQuota := d.Get("daily_memory_time_quota").(int)
 	t := d.Get("tags").(map[string]interface{})
@@ -286,6 +287,8 @@ func resourceFunctionAppSlotCreate(d *pluginsdk.ResourceData, meta interface{})
 			HTTPSOnly:            utils.Bool(httpsOnly),
 			DailyMemoryTimeQuota: utils.Int32(int32(dailyMemoryTimeQuota)),
 			SiteConfig:           &siteConfig,
+			// FORK: Add ClientAffinityEnabled field
+			ClientAffinityEnabled: utils.Bool(clientAffinityEnabled),
 		},
 	}
 
@@ -349,6 +352,7 @@ func resourceFunctionAppSlotUpdate(d *pluginsdk.ResourceData, meta interface{})
 	}
 	appServicePlanID := d.Get("app_service_plan_id").(string)
 	enabled := d.Get("enabled").(bool)
+	clientAffinityEnabled := d.Get("client_affinity_enabled").(bool)
 	httpsOnly := d.Get("https_only").(bool)
 	dailyMemoryTimeQuota := d.Get("daily_memory_time_quota").(int)
 	t := d.Get("tags").(map[string]interface{})
@@ -386,6 +390,8 @@ func resourceFunctionAppSlotUpdate(d *pluginsdk.ResourceData, meta interface{})
 			HTTPSOnly:            utils.Bool(httpsOnly),
 			DailyMemoryTimeQuota: utils.Int32(int32(dailyMemoryTimeQuota)),
 			SiteConfig:           &siteConfig,
+			// FORK: Add ClientAffinityEnabled field
+			ClientAffinityEnabled: utils.Bool(clientAffinityEnabled),
 		},
 	}
 
